### PR TITLE
Support AAD in AES-GCM

### PR DESCRIPTION
**Pull request by sergio-nsk**

### CODE REVIEW

File: mz_crypt_apple.c [hunk 1]
```diff
@@ -25,6 +26,7 @@ enum {
 
 CCCryptorStatus CCCryptorGCMReset(CCCryptorRef cryptorRef);
 CCCryptorStatus CCCryptorGCMAddIV(CCCryptorRef cryptorRef, const void *iv, size_t ivLen);
+CCCryptorStatus CCCryptorGCMaddAAD(CCCryptorRef cryptorRef, const void *aData, size_t aDataLen);
```
Comment (nmoinvaz) [inline mz_crypt_apple.c:29]: Interesting, looks like this wrong cased name. But macOS docs, show
image
But odd that they both show the same macOS and iPhone minimum versions. Perhaps it was an internal name?

Comment (nmoinvaz) [reply mz_crypt_apple.c:29]: Let's change to correct case since it is the same `AVAILABLE_STARTING` version for both functions.

**Comment by nmoinvaz** [comment]
I have removed the calls to `AES_*` in a separate commit. Some changes were missing from this PR, and it should be in a separate commit anyways. So this PR will need rebase. Sorry

**Comment by sergio-nsk** [comment]
> I have removed the calls to `AES_*` in a separate commit. Some changes were missing from this PR, and it should be in a separate commit anyways. So this PR will need rebase. Sorry

You forgot to add `!aes-ctx` in parameter checks in `*encrypt*` and `*decrypt*` functions.

**Pull request merged**